core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -940,23 +974,54 @@ public Void visitValues(ValuesNode node, Void context)
                 }
                 return null;
             }
-            List<String> rows = node.getRows().get().stream()
-                    .map(row -> {
-                        if (row instanceof Row) {
-                            return ((Row) row).getItems().stream()
-                                    .map(PlanPrinter::unresolveFunctions)
-                                    .map(Expression::toString)
-                                    .collect(joining(", ", "(", ")"));
-                        }
-                        return unresolveFunctions(row).toString();
-                    })
-                    .collect(toImmutableList());
-            for (String row : rows) {
-                nodeOutput.appendDetailsLine(row);
+
+            if (verbose) {
+                List<String> rows = node.getRows().get().stream()
+                        .map(this::expressionRepresentation)
+                        .collect(toImmutableList());
+                for (String row : rows) {
+                    nodeOutput.appendDetailsLine(row);
+                }
+            }
+            else {
+                // non-existent rows should not be rendered
+                if (node.getRows().get().isEmpty()) {
+                    return null;
+                }
+                int rowCount = node.getRows().get().size();
+                int valuesLimit = min(VALUES_TRUNCATION_ROW_COUNT, rowCount);
+                // Print first (limit - 1) elements, followed by last element
+                // to provide a readable summary of the contents
+                Stream<String> prefix = Stream.concat(
+                        node.getRows().get().subList(0, valuesLimit - 1)
+                                .stream()
+                                .map(this::expressionRepresentation),
+                        valuesLimit < rowCount ? Stream.of("...") : Stream.of());
+
+                // If truncate, we should print rowCount.
+                if (valuesLimit < rowCount) {
+                    prefix = Stream.concat(Stream.of(format("Row count: %s", rowCount)), prefix);
+                }
```

Comment:
I think it is reasonable either way. It also depends on what is more important information, if overall row count is printed then skipped row count can be derived from it and vice versa. I'm leaning slightly towards overall row count instead of skipped row count.